Fast follows for Unit Testing (#4975)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-properties/unit-test-input.md

---
title: "Input for unit tests"
sidebar_label: "Input"
---

> Inputs in unit testing are used to reference a specific model or source for the test:

Reviewer: ```suggestion
You can use inputs in unit testing to narrow the test to a specific model or source using these guidelines:
```

Author: I like to avoid "You can" as much as possible in instructional and reference documentation to make directions and guidance sound firm. How about:
```suggestion
Use inputs in your unit tests to reference a specific model or source for the test:
```

> - `input:` string that represents a `ref` or `source` call:

Reviewer: ```suggestion
- Use a string that represents a `ref` or `source` call. For example:
```

Author: We don't highlight the `input` before this sentence, so I think we should leave it in some fashion. How about
```suggestion
-  For `input:`, use a string that represents a `ref` or `source` call:
```

    - `ref('my_model')` or `ref('my_model', v='2')` or `ref('dougs_project', 'users')`
    - `source('source_schema', 'source_name')`

> - `input:` is optional for seeds:

Reviewer: ```suggestion
- Optionally use for seeds:
```

    - If you don’t supply an input for a seed, we will use the seed _as_ the input.
    - If you do supply an input for a seed, we will use that input instead.

> - You can also have “empty” inputs, by setting rows to an empty list `rows: []`

Reviewer: ```suggestion
- Use “empty” inputs by setting rows to an empty list `rows: []`
```

> ```yml

Reviewer: Does this belong in the project YAML?

Author: In the /docs unit test doc we specify that Unit tests must be defined in a YML file in the models/ directory.
